src/Slim/Middleware/Session.php

```diff
@@ -43,6 +43,7 @@ public function __construct($settings = [])
             'httponly'    => false,
             'name'        => 'slim_session',
             'autorefresh' => false,
+            'csrf'        => false
```

Comment:
This is my code:

```php
        $this->app->add(new \Slim\Middleware\Session([
            'name' => 'iwsession',
            'autorefresh' => true,
            'lifetime' => '8 hour',
            "csrf" => true
        ]));
        $this->container["helper"] = function (){
            return new Helper();
        };

        $this->container['csrf'] = function () {
            $guard = new Guard();
            $guard->setFailureCallable(function ($request, $response, $next) {
                $request = $request->withAttribute("csrf_result", 'FAILED');

                return $next($request, $response);
            });
            return $guard;
        };

        $this->app->add($this->container->get('csrf'));

```

There is `$this->app` because it's inside a class, but it's the same out of the class, too.

If I don't put `'csrf' => true` I've got this error: `PHP Fatal error:  Uncaught RuntimeException: CSRF middleware failed. Session not found.`
But if I put `'csrf' => true`, starting the session in the constructor, all is right.

That was my entire problem, temporarily solved with the pull request

Bye